app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,50 @@
+class CommentsController < ApplicationController
+
+  def create
+    #if its on a post
+    if params[:post_id]
+      @post = Post.find(params[:post_id])
+      parent_id = params[:post_id]
+      @user = @post.user
+      type = "Post"
+    #if its on a photo
+    else
+      @photo = Photo.find(params[:photo_id])
+      @user = @photo.user
+      parent_id = params[:photo_id]
+      type = "Photo"
+    end
+    @comment = Comment.new(:commentable_id => parent_id.to_i,
+                          :commentable_type => type,
+                          :user_id => current_user.id,
+                          :body => params[:comment][:body])
+    if @comment.save
+      User.delay(run_at: 5.seconds.from_now).send_comment_email(@user.id, current_user.id, type)
+      if type == "Photo"
+        redirect_to user_photo_path(@user, @photo)
+      elsif type == "Post"
+        redirect_to user_posts_path(@user)
+      end
+    else
+      redirect_to user_posts_path(@user)
+    end
+  end
```

Comment:
This method looks a bit hairy. Might be worth pulling some of it out into a model or at least putting it in a private helper in the controller.